hub/api/tests/test_api.py

```diff
@@ -236,6 +237,35 @@ def test_compute_slices(memory_ds):
     _check_tensor(ds.data[:, :][0][(0, 1, 2), 0][1], data[:, :][0][(0, 1, 2), 0][1])
 
 
+def test_length_slices(memory_ds):
+    ds = memory_ds
+    data = np.array([1, 2, 3, 9, 8, 7, 100, 99, 98, 99, 101])
+    ds.create_tensor("data")
+    ds.data.extend(data)
+
+    assert len(ds) == 11
+    assert len(ds[0]) == 1
+    assert len(ds[0:1]) == 1
+    assert len(ds[0:0]) == 0
+    assert len(ds[1:10]) == 9
+    assert len(ds[1:7:2]) == 3
+    assert len(ds[1:8:2]) == 4
+    assert len(ds[1:9:2]) == 4
+    assert len(ds[1:10:2]) == 5
+    assert len(ds[[0, 1, 5, 9]]) == 4
+
+    assert len(ds.data) == 11
+    assert len(ds.data[0]) == 1
+    assert len(ds.data[0:1]) == 1
+    assert len(ds.data[0:0]) == 0
+    assert len(ds.data[1:10]) == 9
+    assert len(ds.data[1:7:2]) == 3
+    assert len(ds.data[1:8:2]) == 4
+    assert len(ds.data[1:9:2]) == 4
+    assert len(ds.data[1:10:2]) == 5
+    assert len(ds.data[[0, 1, 5, 9]]) == 4
```

Comment:
can you add tests for shape properties? like `shape` and `shape_interval`. you can either add them here or in the `test_shape_property` method

```python
ds.data[0].shape
ds.data[1:5].shape
ds.data[1].shape_interval
# etc.
```